bughouse_console/src/server_config.rs

```diff
@@ -37,17 +46,18 @@ pub enum AuthOptions {
     },
 }
 
-#[derive(Debug, Eq, PartialEq, Serialize, Deserialize)]
+#[derive(Debug, Eq, PartialEq, Serialize, Deserialize, Clone)]
 pub enum SessionOptions {
     NoSessions,
 
-    // Sessions terminate on server termination.
-    WithNewRandomSecret,
-
-    // Allows for sessions that survive server restart.
-    // TODO: Support persistent sessions.
-    #[allow(dead_code)]
-    WithSecret(Vec<u8>),
+    WithSessions {
+        // When the secret is preserved, client-side sessions survive server
+        // restarts. When Random is used, or the secret changes,
+        // the sessions are terminated.
+        secret: StringSource,
+        //#[serde(serialize_with = "serialize_duration", deserialize_with = "deserialize_duration")]
```

Comment:
What happens if you don't specify `serialize_with`/`deserialize_with`? Does serde use some non-human-readable format?

Reply:
That's just a comment I forgot to remove. The history behind it is, `time::Duration` doesn't implement serialize/deserialize (at least by default.. now I checked, and it does if I enable the feature), so I was going to implement human-readable format manually.

Went with `std::time::Duration`, see the yaml file for the example of how it's serialized.